### ISSUE TITLE
Generation of associated const array of sized EnumCount

### ISSUE THREAD

**Issue by ratmice** [Peternator7/strum#198]
This is just a (perhaps silly) thought, it is pretty easy to collect into a vec or slice I suppose, with
`let foo: Vec<Foo> = Foo::iter().collect();`

But it occurred to me that because of the non-const trait functions on iterator you cannot 
`let foo: [Foo; Foo::COUNT]; = Foo::iter().collect();`

It might be nice if there was a way to generate an associated const array, perhaps named `Foo::ITEMS` such that essentially the existing `EnumIter` for `Foo::iter()` would be `Foo::ITEMS.iter()`

As I said, i'm not doing anything which needs it, but similar to #96 where the `fn count()` was changed to be an associated const.

**Comment by ratmice** [comment]
Of course, I just realized that since `Default::default()` isn't const this probably be feasible...

**Comment by Peternator7** [comment]
Alas, you are correct, without const trait methods, this would be difficult to do in the general case. 

I'm going to leave it open because I like the idea, but I don't think we can really do anything here without rustc support.

**Comment by dzmitry-lahoda** [comment]
seems we still have no const traits?

would generating without traits work?
